theseus/embodied/kinematics/tests/test_urdf_model.py

```diff
@@ -39,21 +56,52 @@ def test_forward_kinematics_seq(robot_model, dataset):
     for joint_state, ee_pose_target in zip(
         dataset["joint_states"], dataset["ee_poses"]
     ):
-        ee_se3_target = SE3(x_y_z_quaternion=ee_pose_target)
+        ee_se3_target = th.SE3(x_y_z_quaternion=ee_pose_target)
         ee_se3_computed = robot_model.forward_kinematics(joint_state)[ee_name]
 
-        assert torch.allclose(
-            ee_se3_target.local(ee_se3_computed), torch.zeros(6)
-        )
+        assert torch.allclose(ee_se3_target.local(ee_se3_computed), torch.zeros(6))
 
 
 def test_forward_kinematics_batched(robot_model, dataset):
     ee_name = dataset["ee_name"]
 
-    ee_se3_target = SE3(x_y_z_quaternion=dataset["ee_poses"])
+    ee_se3_target = th.SE3(x_y_z_quaternion=dataset["ee_poses"])
     ee_se3_computed = robot_model.forward_kinematics(dataset["joint_states"])[ee_name]
 
     assert torch.allclose(
         ee_se3_target.local(ee_se3_computed),
         torch.zeros(dataset["num_data"], 6),
     )
+
+
+def test_jacobian(robot_model, dataset):
+    ee_name = dataset["ee_name"]
+
+    for joint_state, ee_pose_target in zip(
+        dataset["joint_states"], dataset["ee_poses"]
+    ):
+        ee_se3_target = th.SE3(x_y_z_quaternion=ee_pose_target)
+
+        def fk_func(x):
+            ee_se3_output = robot_model.forward_kinematics(x)[ee_name]
+            return ee_se3_target.local(ee_se3_output)
+
+        # Compute autograd jacobian (autograd joint-to-cost jacobian)
+        jacobian_autograd = torch.autograd.functional.jacobian(
+            fk_func, joint_state
+        ).squeeze()
+
+        # Compute analytical jacobian
+        # (analytical manipulator jacobian * autograd ee-to-cost jacobian)
+        jacobian_m_analytical = torch.cat(
```

Comment:
Directly computing the jacobian of `local` yields the jacobian in the local frame instead of the world frame. The current implementation is from a discussion with Taosha, but is still not working out currently. I am still looking into ways to debug this and make the math work.

The jacobian implementation shouldn't be too hard to do.

Reply:
Ah good idea on adding native (optional) Jacobain call to `forward_kinematics` like we have for other functions.

Reply:
Jacobian tests now working